code/game/gamemodes/changeling/powers/transform.dm

```diff
@@ -7,17 +7,138 @@
 	req_human = 1
 	max_genetic_damage = 3
 
+//changeling clothing items
+/obj/item/clothing/head/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/clothing/gloves/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/clothing/head/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/clothing/mask/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/clothing/shoes/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/clothing/suit/changeling
+	name = "flesh"
+	flags = NODROP
+	allowed = list(/obj/item/changeling)
+
+/obj/item/clothing/under/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/changeling
+	name = "flesh"
+	flags = NODROP
+	slot_flags = SLOT_BELT | SLOT_BACK
+
 //Change our DNA to that of somebody we've absorbed.
 /obj/effect/proc_holder/changeling/transform/sting_action(mob/living/carbon/human/user)
 	var/datum/changeling/changeling = user.mind.changeling
-	var/datum/dna/chosen_dna = changeling.select_dna("Select the target DNA: ", "Target DNA")
+	var/datum/changelingprofile/chosen_prof = changeling.select_dna("Select the target DNA: ", "Target DNA")
 
-	if(!chosen_dna)
+	if(!chosen_prof)
 		return
 
-	user.dna = chosen_dna
-	user.real_name = chosen_dna.real_name
+	user.dna = chosen_prof.dna
+	user.real_name = chosen_prof.name
 	hardset_dna(user, null, null, null, null, chosen_dna.species.type, chosen_dna.features)
+	var/list/slots = list("head", "wear_mask", "back", "wear_suit", "w_uniform", "shoes", "belt", "gloves", "glasses", "ears", "wear_id", "s_store")
+
+	//im so sorry

```

Comment:
Jesus, cant you loop through get_inventory_slots() or another similar inventory proc?